util/signing/src/lib.rs

```diff
@@ -0,0 +1,48 @@
+/// A collection of bytes.
+#[derive(Debug, Clone)]
+pub struct Bytes(pub Vec<u8>);
+
+/// A signature.
+#[derive(Debug, Clone)]
+pub struct Signature(pub Bytes);
+
+/// A public key.
+#[derive(Debug, Clone)]
+pub struct PublicKey(pub Bytes);
+
+/// Version of a key.
+/// Default mean the current key.
+#[derive(Debug, Clone, Default)]
+pub struct KeyVersion(pub String);
+
+/// Id that identify a Key.
+#[derive(Debug, Clone)]
+pub struct KeyId(pub String);
+
+pub struct SigningService;
+
+impl SigningService {
+	/// Create the service with environment variable.
+	pub fn try_from_env() -> Result<Self, anyhow::Error> {
```

Comment:
If `SigningService` (or `Hsm` in the earlier demo PRs) is to be a trait, this does not belong in the abstraction. We potentially want the dynamic dispatch to be feasible, and the constructor APIs should belong with the specific backends.